### ISSUE TITLE
Release primitive-0.6.4.0

### ISSUE THREAD

**Issue by andrewthad** [haskell/primitive#154]
I think commit e1854d7b691a3856be20cb509ca0f14bb72e1801 is a good candidate for a release.

cc @cartazio

**Comment by cartazio** [comment]
I'd like to merge in the MVar work with 1-2 edits after merge plus adding some cleanup to the parts of docs where we talk about affine, but yes, otherwise agreed

or screw it, lets just do a release with that commit and we can do another release soon after

sounds good to me :) 

i'll do it this AM :)

**Comment by cartazio** [comment]
(i do wanna get a Mvar capable release out soon after though, cause the multi threaded ART data structure will be lovely to write on top :) )

**Comment by andrewthad** [comment]
If you've got a use for the `MVar` stuff, I'm happy to wait on that. I'm not in any real hurry.

**Comment by andrewthad** [comment]
I just realized there's some stuff for `UnliftedArray` that I want to add (folds and map).

**Comment by cartazio** [comment]
cool.

i assume the obvious implementations for this release then cleanup later?

**Comment by cartazio** [comment]
https://github.com/haskell/primitive/commit/49440b267748589079ce04f65a3af6da356ba3a2 should be the current base / root of the release right?

(i may try to poke at some documentation ideas, but i'll at least cut a package candidate today even though i wanna poke at those bits :) )

**Comment by andrewthad** [comment]
Yes, that commit would be a good release. I'd still like to see something like  https://github.com/haskell/primitive/pull/68 happen for documentation purposes, but it would be nice if it gave such a disclaimer for all of the `Ord` instances, not just for `ByteArray`. This would provide greater flexibility to change them in the future.

**Comment by RyanGlScott** [comment]
Would you be willing to provide a PR on top of #68? I'm not confident that I'd choose the best wording, but perhaps you could.

**Comment by cartazio** [comment]
yeah, throw it our way and thats a fast review for all of us

@RyanGlScott should we give @andrewthad commit bit at this point as long as he talks iwth folks before firing missiles?

**Comment by RyanGlScott** [comment]
Agreed. I've send @andrewthad an invite.

**Comment by andrewthad** [comment]
Working on the `Ord` disclaimer.

**Comment by andrewthad** [comment]
New PR for `Ord` docs is up: https://github.com/haskell/primitive/pull/161. I'm not terribly confident of the wording, but any suggestions are welcome on the PR.

**Comment by cartazio** [comment]
once https://github.com/haskell/primitive/commit/85d6c2f07b88d6e97f08dc84e4beca4a0f77c377 is green on travis i'll cut the release :)

**Comment by andrewthad** [comment]
Awesome. I'm stoked to get rid of all the git submodules and `cabal.project` overrides of primitive that have been polluting everything I've been working on for the last month.

**Comment by cartazio** [comment]
https://travis-ci.org/haskell/primitive all green,

**Comment by cartazio** [comment]
https://hackage.haskell.org/package/primitive-0.6.4.0/candidate is the package candidate. please test/eyeball before i promote it and push a tag and commit to master

**Comment by cartazio** [comment]
https://github.com/haskell/primitive/commit/2e0e555fafe0be5debb06350832ce1a79d96afe4 is the changes i made

**Comment by andrewthad** [comment]
This has been released. Closing.
